rust/operator-binary/src/controller.rs

```diff
@@ -459,12 +462,19 @@ pub async fn reconcile_nifi(nifi: Arc<NifiCluster>, ctx: Arc<Ctx>) -> Result<Act
         })?;
 
     let nifi_authentication_config = NifiAuthenticationConfig::try_from(
-        resolve_authentication_classes(client, &nifi.spec.cluster_config.authentication)
+        AuthenticationClassResolved::from(&nifi, client)
             .await
             .context(FailedResolveNifiAuthenticationConfigSnafu)?,
     )
     .context(InvalidNifiAuthenticationConfigSnafu)?;
 
+    if let NifiAuthenticationConfig::Oidc { .. } = nifi_authentication_config {
+        tracing::info!("Checking for OIDC admin password configuration");
```

Comment:
Thanks for adding tracing! I think I would prefer moving this into `check_or_generate_oidc_admin_password` and reducing it to debug to reduce noise, as we already have `tracing::info!("No existing oidc admin password secret found, generating new one");` in the important case it get's created.